community/lucene-index/src/main/java/org/neo4j/kernel/api/impl/schema/LuceneDocumentStructure.java

```diff
@@ -150,6 +150,14 @@ public static Query newWildCardStringQuery( String searchFor )
         return new WildcardQuery( term );
     }
 
+    public static Query newSuffixStringQuery( String suffix )
+    {
+        String searchTerm = QueryParser.escape( suffix );
+        Term term = new Term( ValueEncoding.String.key(), "*" + searchTerm );
```

Comment:
For an optimised version, we would need to store the data reversed as well. That is much more expensive though - it makes the indexes slower and bigger to update. This change is simply using the data structures already in place.

I think we should consider double-storing things as an option when creating indexes, so the user gets to choose if they want fast reads or fast writes.